### PR TITLE
Fix enum naming

### PR DESCRIPTION
**Pull request by kelnos**
Addresses https://github.com/twilio/guardrail/issues/264

This also cleans up some deficiencies with our case converters, namely not handling spaces as separators in the original identifier, and not handling numbers as the initial character in a grouping when converting to pascal case.

**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/JacksonGenerator.scala [hunk 1]
```diff
@@ -103,7 +103,7 @@ object JacksonGenerator {
           case (value, termName, _) =>
             new EnumConstantDeclaration(
               new NodeList(),
-              new SimpleName(termName.getIdentifier.toSnakeCase.toUpperCase(Locale.US)),
```
Comment (blast-hardcheese) [inline modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/JacksonGenerator.scala]: This is in "Add FormatEnumName term to ScalaTerms", but it doesn't appear to be doing that, what's going on here?

File: modules/sample/src/main/resources/issues/issue264.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      responses:
+        204:
+          description: OK
+components:
+  schemas:
+    FooEnum:
+      type: string
+      enum:
+        - foo-bar-1
+        - baz-quux-2-flurb
```
Comment (blast-hardcheese) [inline modules/sample/src/main/resources/issues/issue264.yaml]: You mentioned "not handling numbers as the initial character in a grouping", but what if the numbers are the initial character in the entire enumeration?

File: modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/JacksonGenerator.scala (hunk 1 above)
Comment (kelnos) [reply modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/JacksonGenerator.scala]: Without this change it'll generate very wrong names with the addition of a (correct) `FormatEnumName` for Java.  In general I like to make each commit as correct as possible.

File: modules/sample/src/main/resources/issues/issue264.yaml (hunk 2 above)
Comment (kelnos) [reply modules/sample/src/main/resources/issues/issue264.yaml]: Welp, regardless, it'll be an invalid name in Java at least.

File: modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/JacksonGenerator.scala (hunk 1 above)
Comment (blast-hardcheese) [reply modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/JacksonGenerator.scala]: Ah, due to the double snake case. Makes sense. Thanks!

**Pull request merged**